gammapy/modeling/models/spectral.py

```diff
@@ -1533,6 +1536,105 @@ def from_region_map(cls, map, **kwargs):
 
 
+class TemplateNDSpectralModel(SpectralModel):
+    """A model generated from a ND array where extra dimensions define the parameter space.
+
+    For more information see :ref:`templateND-spectral-model`.
+
+    Parameters
+    ----------
+    map : `~gammapy.maps.RegionNDMap`
+        Map template. 
+    meta : dict, optional
+        Meta information, meta['filename'] will be used for serialization
+    interp_kwargs : dict
+        Interpolation keyword arguments passed to `gammapy.maps.Map.interp_by_coord`.
+        Default arguments are {'interp': 'linear', 'fill_value': 0}.
+    """
+
+    tag = ["TemplateNDSpectralModel", "templateND"]
+
+    def __init__(self, map, interp_kwargs=None, meta=None, filename=None):
+        self._map = map.copy()
+        self.meta = dict() if meta is None else meta
+        if filename is not None:
+            filename = str(make_path(filename))
+        self.filename = filename
+
+        points_scale = [
+            "lin",
+            "lin",
+            "log",
+        ]
+        parameters = []
+        for axe in map.geom.axes:
+            if axe.name not in ["energy_true", "energy"]:
+                center = (axe.bounds[1] + axe.bounds[0]) / 2
+                parameter = Parameter(
+                    name=axe.name,
+                    value=center,
+                    unit=axe.unit,
+                    scale_method="scale10",
+                    min=axe.bounds[0],
+                    max=axe.bounds[-1],
+                    interp="lin",
+                )
+                points_scale.append("lin")
+                parameters.append(parameter)
+        self.default_parameters = Parameters(parameters)
+
+        interp_kwargs = interp_kwargs or {}
+        interp_kwargs.setdefault("values_scale", "log")
+        interp_kwargs.setdefault("points_scale", points_scale)
+        self._interp_kwargs = interp_kwargs
+        super().__init__()
+
+    @property
+    def map(self):
+        """Template map  (`~gammapy.maps.RegionNDMap`)"""
+        return self._map
+
+    def evaluate(self, energy, **kwargs):
+        lon = self.map.geom.center_skydir.data.lon.deg
+        lat = self.map.geom.center_skydir.data.lat.deg
+        coord = {"lon": lon, "lat": lat, "energy_true": energy}
+        coord.update(kwargs)
+
+        pixels = list(self.map.geom.coord_to_pix(coord))
+        # TODO : pixels in lon/lat are nan at center coord, why ?
+        pixels[0][np.isnan(pixels[0])] = 0
+        pixels[1][np.isnan(pixels[1])] = 0
+
+        val = self.map.interp_by_pix(pixels, **self._interp_kwargs)
+        return u.Quantity(val, self.map.unit, copy=False)
+
+        super().__init__()
+
+    def write(self, overwrite=False):
+        if self.filename is None:
+            raise IOError("Missing filename")
+        elif os.path.isfile(self.filename) and not overwrite:
+            log.warning("Template file already exits, and overwrite is False")
```

Comment:
Maybe raise an `IOError` here. Or is it to avoid exceptions when exporting `Models` with template files?
In the latter case, maybe adding a test would be useful.

Reply:
if `os.path.isfile(self.filename) and not overwrite` then `.write` will return an error but I prefer to avoid